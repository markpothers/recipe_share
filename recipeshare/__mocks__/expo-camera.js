

export const Camera = {
	requestCameraPermissionsAsync: () => {
		return Promise.resolve({granted: true})
	}
}
