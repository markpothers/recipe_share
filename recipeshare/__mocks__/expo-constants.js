export default {
	manifest: {
		ios: {
			buildNumber: 22
		},
		android: {
			versionCode: 22
		}
	}
}
