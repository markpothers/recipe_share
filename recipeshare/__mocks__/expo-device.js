let deviceType = 1

export const setMockDeviceType = (type) => {
    deviceType = type
}

export const getDeviceTypeAsync = jest.fn(() => {
    // console.warn('Im here!')
    return new Promise(resolve => {
        resolve(deviceType)
    })
});