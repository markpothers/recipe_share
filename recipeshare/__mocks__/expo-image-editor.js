import React, { useEffect } from "react";
import { View } from "react-native";

export const ImageEditor = (props) => {
	useEffect(()=>{
		props.onEditingComplete({
			cancelled: false,
			uri: props.imageUri,
		})
	}, [props])
	return <View testID={"mockImageEditor"}></View>;
};
