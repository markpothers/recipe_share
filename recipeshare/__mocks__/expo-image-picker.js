

export const launchCameraAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockCameraUri"}]
	})
}

export const launchImageLibraryAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockLibraryUri"}]
	})
}
