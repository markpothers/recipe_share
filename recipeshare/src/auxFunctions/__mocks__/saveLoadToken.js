export function saveToken () {
	console.log("mock save token!!!!!")
	return null
}

export function loadToken () {
	return null
}
