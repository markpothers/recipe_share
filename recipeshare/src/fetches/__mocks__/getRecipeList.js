import { mockRecipeList } from '../../../__mocks__/recipeList/mockRecipeList'

export const getRecipeList = () => {
    // console.warn('mock fetch was called')
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
