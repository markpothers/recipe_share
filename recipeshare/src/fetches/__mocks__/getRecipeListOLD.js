import { mockRecipeList } from "../../../__mocks__/mockRecipeList"

export const getRecipeList = () => {
    console.warn("mock fetch was called")
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
