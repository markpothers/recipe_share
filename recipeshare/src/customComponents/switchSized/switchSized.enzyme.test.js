/**
 * @jest-environment jsdom
 */
import React from 'react';
import { mount } from 'enzyme'
import { act } from 'react-dom/test-utils';
import { createSerializer } from 'enzyme-to-json';
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
import toJson from 'enzyme-to-json';

import { Switch, Platform } from 'react-native'
import SwitchSized from './switchSized.js'
import { setMockDeviceType } from '../../../__mocks__/expo-device'

describe('SwitchSized', () => {

	let component
	let mockChangeFunction

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	})

	beforeEach(async () => {
		// console.log('runs before every test')
		mockChangeFunction = jest.fn()
	})

	afterEach(() => {
		// console.log('runs after each test')
	})

	afterAll(() => {
		// console.log('runs after all tests have completed')
	})

	test('can be rendered on an iPhone with reduced size switches where switch is on', async () => {
		setMockDeviceType(1) //phone
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on an iPhone with reduced size switches where switch is off', async () => {
		setMockDeviceType(1)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on an android switch is off', async () => {
		Platform.OS = 'android'
		setMockDeviceType(1)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on an android where switch is on', async () => {
		Platform.OS = 'android'
		setMockDeviceType(1)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on iPad with regular switch where switch is on', async () => {
		setMockDeviceType(2) //iPad
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered on iPad with regular switch where switch is off', async () => {
		setMockDeviceType(2)
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let json = toJson(component)
		expect(json).toMatchSnapshot();
	})

	test('can be rendered and switched off', async () => {
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let testSwitch = component.find(Switch)
		expect(testSwitch.length).toEqual(1)
		testSwitch.props().onValueChange()
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(1)
	})

	test('can be rendered and switched on', async () => {
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let testSwitch = component.find(Switch)
		expect(testSwitch.length).toEqual(1)
		testSwitch.props().onValueChange()
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(1)
	})

	test('can be rendered and switched on and off', async () => {
		await act(async () => {
			component = await mount(
				<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
			)
		})
		component.setProps({})
		let testSwitch = component.find(Switch)
		expect(testSwitch.length).toEqual(1)
		testSwitch.props().onValueChange()
		testSwitch.props().onValueChange()
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(2)
	})

});
