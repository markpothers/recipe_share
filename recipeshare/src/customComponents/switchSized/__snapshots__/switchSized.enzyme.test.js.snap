// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSized can be rendered on an android switch is off 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={false}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        null,
        null,
      ]
    }
    thumbColor="#eaeaea"
    trackColor={
      Object {
        "false": "#64715599",
        "true": "#5c8a5199",
      }
    }
    value={false}
  >
    <AndroidSwitch
      accessibilityRole="switch"
      enabled={true}
      on={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          null,
        ]
      }
      thumbTintColor="#eaeaea"
      trackColorForFalse="#64715599"
      trackColorForTrue="#5c8a5199"
      trackTintColor="#64715599"
    >
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            null,
            null,
          ]
        }
        thumbTintColor="#eaeaea"
        trackColorForFalse="#64715599"
        trackColorForTrue="#5c8a5199"
        trackTintColor="#64715599"
      />
    </AndroidSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on an android where switch is on 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={true}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        null,
        null,
      ]
    }
    thumbColor="#4b7142"
    trackColor={
      Object {
        "false": "#64715599",
        "true": "#5c8a5199",
      }
    }
    value={true}
  >
    <AndroidSwitch
      accessibilityRole="switch"
      enabled={true}
      on={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          null,
        ]
      }
      thumbTintColor="#4b7142"
      trackColorForFalse="#64715599"
      trackColorForTrue="#5c8a5199"
      trackTintColor="#5c8a5199"
    >
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={true}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            null,
            null,
          ]
        }
        thumbTintColor="#4b7142"
        trackColorForFalse="#64715599"
        trackColorForTrue="#5c8a5199"
        trackTintColor="#5c8a5199"
      />
    </AndroidSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on an iPhone with reduced size switches where switch is off 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={false}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scaleX": 0.7,
            },
            Object {
              "scaleY": 0.7,
            },
          ],
        },
        null,
      ]
    }
    thumbColor={null}
    trackColor={
      Object {
        "false": null,
        "true": "#5c8a5199",
      }
    }
    value={false}
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#5c8a5199"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 0.7,
                },
                Object {
                  "scaleY": 0.7,
                },
              ],
            },
            null,
          ],
        ]
      }
      thumbTintColor={null}
      tintColor={null}
      value={false}
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#5c8a5199"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Array [
              Object {
                "transform": Array [
                  Object {
                    "scaleX": 0.7,
                  },
                  Object {
                    "scaleY": 0.7,
                  },
                ],
              },
              null,
            ],
          ]
        }
        thumbTintColor={null}
        tintColor={null}
        value={false}
      />
    </RCTSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on an iPhone with reduced size switches where switch is on 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={true}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scaleX": 0.7,
            },
            Object {
              "scaleY": 0.7,
            },
          ],
        },
        null,
      ]
    }
    thumbColor={null}
    trackColor={
      Object {
        "false": null,
        "true": "#5c8a5199",
      }
    }
    value={true}
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#5c8a5199"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "scaleX": 0.7,
                },
                Object {
                  "scaleY": 0.7,
                },
              ],
            },
            null,
          ],
        ]
      }
      thumbTintColor={null}
      tintColor={null}
      value={true}
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#5c8a5199"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Array [
              Object {
                "transform": Array [
                  Object {
                    "scaleX": 0.7,
                  },
                  Object {
                    "scaleY": 0.7,
                  },
                ],
              },
              null,
            ],
          ]
        }
        thumbTintColor={null}
        tintColor={null}
        value={true}
      />
    </RCTSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on iPad with regular switch where switch is off 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={false}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        null,
        null,
      ]
    }
    thumbColor="#eaeaea"
    trackColor={
      Object {
        "false": "#64715599",
        "true": "#5c8a5199",
      }
    }
    value={false}
  >
    <AndroidSwitch
      accessibilityRole="switch"
      enabled={true}
      on={false}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          null,
        ]
      }
      thumbTintColor="#eaeaea"
      trackColorForFalse="#64715599"
      trackColorForTrue="#5c8a5199"
      trackTintColor="#64715599"
    >
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={false}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            null,
            null,
          ]
        }
        thumbTintColor="#eaeaea"
        trackColorForFalse="#64715599"
        trackColorForTrue="#5c8a5199"
        trackTintColor="#64715599"
      />
    </AndroidSwitch>
  </Switch>
</SwitchSized>
`;

exports[`SwitchSized can be rendered on iPad with regular switch where switch is on 1`] = `
<SwitchSized
  onValueChange={[MockFunction]}
  value={true}
>
  <Switch
    onValueChange={[MockFunction]}
    style={
      Array [
        null,
        null,
      ]
    }
    thumbColor="#4b7142"
    trackColor={
      Object {
        "false": "#64715599",
        "true": "#5c8a5199",
      }
    }
    value={true}
  >
    <AndroidSwitch
      accessibilityRole="switch"
      enabled={true}
      on={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          null,
        ]
      }
      thumbTintColor="#4b7142"
      trackColorForFalse="#64715599"
      trackColorForTrue="#5c8a5199"
      trackTintColor="#5c8a5199"
    >
      <AndroidSwitch
        accessibilityRole="switch"
        enabled={true}
        on={true}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            null,
            null,
          ]
        }
        thumbTintColor="#4b7142"
        trackColorForFalse="#64715599"
        trackColorForTrue="#5c8a5199"
        trackTintColor="#5c8a5199"
      />
    </AndroidSwitch>
  </Switch>
</SwitchSized>
`;
